packages/arb-validator-core/gobridge/arbClient.go

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobridge
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+)
+
+var errReorgError = errors.New("reorg occured")
+var headerRetryDelay = time.Second * 2
+var maxFetchAttempts = 5
+
+type GoArbClient struct {
+	GoEthClient *goEthdata
+}
+
+func NewEthClient(ethURL string) (*GoArbClient, error) {
+	// call to goEth.go - getGoEth(ethURL)
+	client := GoArbClient{getGoEth(ethURL)}
+
+	return &client, nil
+}
```

Comment:
There's no chance that this function returns an error, so the error type should be removed from its return value.

Reply:
fixed